packages/next-app/src/components/OwnedPool.tsx

```diff
@@ -2,23 +2,46 @@ import { Badge, Button, HStack, Text, VStack } from "@chakra-ui/react";
 
 interface IOwnedPool {
   onClick: () => void;
-  address: string;
-  type: string;
-  ratio?: string;
-  name: string;
+  address?: string;
+  poolType?: string | null | undefined;
+  name?: string | null | undefined;
+  id?: string | null | undefined;
+  tokens?:
+    | {
+        symbol: string;
+        weight?: unknown;
+      }[]
+    | null
+    | undefined;
   isSelected: boolean;
 }
 
+function truncateAddress(address: string | undefined) {
+  if (!address) return;
+  const match = address.match(
+    /^(0x[a-zA-Z0-9]{10})[a-zA-Z0-9]+([a-zA-Z0-9]{4})$/
+  );
+  if (!match) return address;
+  return `${match[1]}…${match[2]}`;
+}
+
 export function OwnedPool({
   onClick,
   address,
-  type,
-  ratio,
+  poolType,
   name,
   isSelected,
+  tokens,
 }: IOwnedPool) {
   const backgroundColor = isSelected ? "blue.800" : "gray.800";
 
+  const poolName =
+    poolType === "Weighted" ? tokens!.map((obj) => obj.symbol).join("/") : name;
```

Comment:
Curious: are weighted pool's names not the same as the trading tuple symbols joined by `/`?

Reply:
The pool names are joined with the token and the weight. I changed it to be just the `token/token/...` and the weights in a bagde component. Here we have an example of the original pool names:
<img width="361" alt="image" src="https://user-images.githubusercontent.com/53192759/223410623-6966e585-046b-4d2b-be72-001973e3d610.png">
Note that I only applied the change to the weighted pools to be closer to our HI-FI